packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +127,66 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot ??= (async () => {
+				this.latestSnapshot = await getLatestSnapshotInfo(
+					this.mc,
+					this.storageAdapter,
+					supportGetSnapshotApi,
+				);
+				this.refreshPendingAttributes();
+			})();
+
 			return { baseSnapshot, version: undefined };
 		}
 	}
 
+	/**
+	 * Updates class snapshot and processedOps with the latest fetched snapshot
+	 *
+	 * @param baseSnapshot -
+	 */
+	private refreshPendingAttributes() {
```

Comment:
this should be renamed to match what it does.

Reply:
I'm bad with names :P hope you like the new one

Reply:
what you changed it to is much better. no need to change unless you want to, but here are some things i do to names things:

-  I generally `try<..>` rather than `<..>Maybe` for function names, where it doesn't always do the thing
- describe the things the functions does in a sentence, then try to simplify. usually keeping nouns and verbs
- use an online thesaurus to find synonyms to words. This can be good to get both variety, and words the fit better